commands.go

```diff
@@ -593,6 +593,20 @@ func DownloadAndDecompressWALFile(pre *S3Prefix, walFileName string, dstLocation
 
 // HandleWALPush is invoked to perform wal-g wal-push
 func HandleWALPush(tarUploader *TarUploader, dirArc string, pre *S3Prefix, verify bool) {
+	archive := &Archive{
+		Prefix:  pre,
+		Archive: aws.String(sanitizePath(tarUploader.server + WalPath + filepath.Base(dirArc) + "." + tarUploader.compressor.FileExtension())),
+	}
+
+	exists, err := archive.CheckExistence()
+	if err != nil {
+		log.Fatalf("%+v\n", err)
+	}
+	if exists {
+		log.Printf("WAL file '%s' already archived, not overwriting", dirArc)
+		return
```

Comment:
Will this return a non-zero exit code? We'll want that so postgres knows the archive_command failed and will keep the WAL file around.

Reply:
It exits with status `1` now instead.